lib/ansible/module_utils/facts/system/service_mgr.py

```diff
@@ -137,6 +137,9 @@ def collect(self, module=None, collected_facts=None):
                 service_mgr_name = 'systemd'
             elif module.get_bin_path('initctl') and os.path.exists("/etc/init/"):
                 service_mgr_name = 'upstart'
+            # FIXME: use translation map when checking symlinks
+            elif os.path.islink('/sbin/init') and os.path.basename(os.readlink('/sbin/init')) == 'runit-init':
```

Comment:
To allow for future growth and other fallbacks
```
elif os.path.islink('/sbin/init'):
   service_mg_name =   os.path.basename(os.readlink('/sbin/init')) 
   if service_mg_name  == 'runit-init':
```

Reply:
was just rethinking this a bit, might be a simpler change in line 100, trap the failure of getting file contents above and do the symlink/basename there and assign to proc_1.

But i fear both are invasive, also the proc_1_map should be used instead of the if I suggested previously.

Reply:
Good point about proc_1_map. Makes sense to make use of that. I've changed it like you suggested, if I understood you correctly. I like the simplicity of the change. Just two lines of code and it is using basename call and the map lookup call that are already there.

Reply:
yep, sorry it took me so long to realize it could be like this, had not looked at full file, when i did it became a much less invasive change.